RecoTracker/TkTrackingRegions/plugins/PixelInactiveAreaFinder.cc

```diff
@@ -0,0 +1,1104 @@
+      lower = zAxisIntersection(pointUpperDetGroupL,pointLowerDetGroupL);
+      // upper intersectionpoint
+      const float pointUpperDetGroupU[2] = {cspanUpper.zSpan.first, cspanUpper.rSpan.first};
+      const float pointLowerDetGroupU[2] = {cspanLower.zSpan.second,  cspanLower.rSpan.second};
+      upper = zAxisIntersection(pointUpperDetGroupU,pointLowerDetGroupU);
+    }else if (cspanUpper.zSpan.first <= cspanLower.zSpan.second && cspanLower.zSpan.first <= cspanUpper.zSpan.second){
+      // lower intersectionpoint, point = {z,r} in cylindrical coordinates
+      const float pointUpperDetGroupL[2] = {cspanUpper.zSpan.second, cspanUpper.rSpan.first};
+      const float pointLowerDetGroupL[2] = {cspanLower.zSpan.first,  cspanLower.rSpan.second};
+      lower = zAxisIntersection(pointUpperDetGroupL,pointLowerDetGroupL);
+      // upper intersectionpoint
+      const float pointUpperDetGroupU[2] = {cspanUpper.zSpan.first, cspanUpper.rSpan.first};
+      const float pointLowerDetGroupU[2] = {cspanLower.zSpan.second,  cspanLower.rSpan.second};
+      upper = zAxisIntersection(pointUpperDetGroupU,pointLowerDetGroupU);
+    }else if (cspanUpper.zSpan.first > cspanLower.zSpan.second){
+      // lower intersectionpoint, point = {z,r} in cylindrical coordinates
+      const float pointUpperDetGroupL[2] = {cspanUpper.zSpan.second, cspanUpper.rSpan.first};
+      const float pointLowerDetGroupL[2] = {cspanLower.zSpan.first,  cspanLower.rSpan.second};
+      lower = zAxisIntersection(pointUpperDetGroupL,pointLowerDetGroupL);
+      // upper intersectionpoint
+      const float pointUpperDetGroupU[2] = {cspanUpper.zSpan.first, cspanUpper.rSpan.second};
+      const float pointLowerDetGroupU[2] = {cspanLower.zSpan.second,  cspanLower.rSpan.first};
+      upper = zAxisIntersection(pointUpperDetGroupU,pointLowerDetGroupU);
+    }else{
+      //something wrong
+      return false;
+    }
+    range = std::pair<float,float>(lower,upper);
+    return true;
+  }
+  bool getZAxisOverlapRangeEndcap(const PixelInactiveAreaFinder::DetGroupSpan & cspanA, const PixelInactiveAreaFinder::DetGroupSpan & cspanB, std::pair<float,float> & range){
+    // While on left hand side of pixel detector
+    PixelInactiveAreaFinder::DetGroupSpan cspanNearer;
+    PixelInactiveAreaFinder::DetGroupSpan cspanFurther;
+    float lower = 0;
+    float upper = 0;
+    if(cspanA.zSpan.first < 0 && cspanB.zSpan.first < 0){
+      if(cspanA.zSpan.second < cspanB.zSpan.first){
+        cspanFurther = cspanA;
+        cspanNearer = cspanB;
+      }else if (cspanB.zSpan.second < cspanA.zSpan.first){
+        cspanFurther = cspanB;
+        cspanNearer = cspanA;
+      }else {
+        //edm::LogPrint("") << "No overlap, same disk propably. Spans:";
+        //Stream ss;
+        //detGroupSpanInfo(cspanA,ss);ss<<std::endl;detGroupSpanInfo(cspanB,ss);ss<<std::endl;
+        //edm::LogPrint("") << ss.str();ss.str(std::string());
+        //edm::LogPrint("") << "**";
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
If this and the other printouts are actually needed, better wrap them in EDM_ML_DEBUG and uncomment.

Reply:
The printouts look like they could be useful also later, so I'll wrap them to EDM_ML_DEBUG.